# **STAT 640 : Homework 1**
## Pavan Malapati
## tq8817


Install the Python Pacakages and import necessary libraries

In [1]:
!pip install mistat
!pip list -v
import math
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import mistat
from scipy import stats

  Preparing metadata (setup.py) ... done
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 13.1/13.1 MB 18.2 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 198.9/198.9 kB 5.9 MB/s eta 0:00:00
  Created wheel for diversipy: filename=diversipy-0.9-py3-none-any.whl size=27699 sha256=fe755025084ede419a23be3541a5ab2396880e4db1e207c20dbd7dd7ba24a643
  Stored in directory: /root/.cache/pip/wheels/62/3e/76/8a7f598707243c5493ee4eeeaafc2fec3ef00d8dafb09e7ff1
Successfully built diversipy
Package                          Version               Location                                Installer
-------------------------------- --------------------- --------------------------------------- ---------
absl-py                          1.4.0                 /usr/local/lib/python3.10/dist-packages pip
accelerate                       0.32.1                /usr/local/lib/python3.10/dist-packages pip
aiohappyeyeballs                 2.4.0                 /usr/local/lib/python3.10/dist-packages pip
aio

## Exercise 1.3

Generate a sequence of 50 random binary numbers (0, 1), when the likelihood of 1 is p using the command binom.rvs(1, p, size=50).

In [16]:
from scipy.stats import binom
import numpy as np

np.random.seed(1)

# Generate and print 50 random samples from a binomial distribution with n=1 and p=0.5
b_s = binom.rvs(1, 0.5, size=50)
print(b_s)

[0 1 0 0 0 0 0 0 0 1 0 1 0 1 0 1 0 1 0 0 1 1 0 1 1 1 0 0 0 1 0 0 1 1 1 0 1
 1 0 1 1 1 0 1 0 0 1 0 0 0]


Do this for the values p = 0.1, 0.3, 0.7, 0.9. Count the number of 1s in these random sequences, by summing up the result sequence.

In [17]:
np.random.seed(1)
for p in (0.1, 0.3, 0.7, 0.9):
  o = binom.rvs(1, p, size=50)
  print(p, sum(o))

0.1 4
0.3 12
0.7 33
0.9 43


The expected values of the sums are 5, 15, 35 and 45.

The outcomes are 4,12,33 and 43 for p = 0.1,0.3,0.7,0.9.Therefore, The difference between my outcome and the expected values are 1,3,2 and 2.

## Exercise 1.4

The following are two sets of measurements of the weight of an object, which correspond to two different weighing instruments. The object has a true weight of 10 kg.

In [18]:
i1 = [9.490950, 10.436813, 9.681357, 10.996083, 10.226101, 10.253741,
10.458926, 9.247097, 8.287045, 10.145414, 11.373981, 10.144389,
11.265351, 7.956107, 10.166610, 10.800805, 9.372905, 10.199018,
9.742579, 10.428091]

print(i1)

i2 = [11.771486, 10.697693, 10.687212, 11.097567, 11.676099,
10.583907, 10.505690, 9.958557, 10.938350, 11.718334,
11.308556, 10.957640, 11.250546, 10.195894, 11.804038,
11.825099, 10.677206, 10.249831, 10.729174, 11.027622]

print(i2)

[9.49095, 10.436813, 9.681357, 10.996083, 10.226101, 10.253741, 10.458926, 9.247097, 8.287045, 10.145414, 11.373981, 10.144389, 11.265351, 7.956107, 10.16661, 10.800805, 9.372905, 10.199018, 9.742579, 10.428091]
[11.771486, 10.697693, 10.687212, 11.097567, 11.676099, 10.583907, 10.50569, 9.958557, 10.93835, 11.718334, 11.308556, 10.95764, 11.250546, 10.195894, 11.804038, 11.825099, 10.677206, 10.249831, 10.729174, 11.027622]


Which instrument seems to be more accurate?

In [21]:
print('mean of instrument 1 : ', np.mean(i1))
print('mean of instrument 2 : ', np.mean(i2))

mean of instrument 1 :  10.03366815
mean of instrument 2 :  10.98302505


After Comparing the means of both instruments, The mean of instrument 1 is more closer to 10 kg so, The Instrument 1 is more accurate.

Which instrument seems to be
more precise?

In [22]:
print('Standard Deviation of instrument 1 : ', np.std(i1))
print('Standard Deviation of Instrument 2 : ', np.std(i2))

Standard Deviation of instrument 1 :  0.848764943262696
Standard Deviation of Instrument 2 :  0.5541593648343835


After Comparing the Standard Deviation of both instruments, The instrument 2 is more precise.

## Exercise 1.22

The following data is the time (in sec.) to get from 0 to 60 mph for a sample of 15 German- and 20 Japanese-made cars.

In [26]:
germanCars = [10, 10.9, 4.8, 6.4, 7.9, 8.9, 8.5, 6.9, 7.1,
5.5, 6.4, 8.7, 5.1, 6.0, 7.5]
japaneseCars = [9.4, 9.5, 7.1, 8.0, 8.9, 7.7, 10.5, 6.5, 6.7,
9.3, 5.7, 12.5, 7.2, 9.1, 8.3, 8.2, 8.5, 6.8, 9.5, 9.7]

          German   Japanese
count  15.000000  20.000000
mean    7.373333   8.455000
std     1.780235   1.589596
min     4.800000   5.700000
25%     6.200000   7.175000
50%     7.100000   8.400000
75%     8.600000   9.425000
max    10.900000  12.500000


Compare and contrast the acceleration times of German- and Japanese-made cars, in terms of their five summary statistics.

In [27]:
germanCars = pd.Series(germanCars)
japaneseCars = pd.Series(japaneseCars)

comp_stat = pd.DataFrame({
  'German': germanCars.describe(),
  'Japanese': japaneseCars.describe(),
})
print(comp_stat)

          German   Japanese
count  15.000000  20.000000
mean    7.373333   8.455000
std     1.780235   1.589596
min     4.800000   5.700000
25%     6.200000   7.175000
50%     7.100000   8.400000
75%     8.600000   9.425000
max    10.900000  12.500000


German-made cars have a quicker average acceleration time (7.37 seconds) compared to Japanese-made cars (8.46 seconds). The range of times for German cars is also wider, with times going from 4.8 to 10.9 seconds, while Japanese cars range from 5.7 to 12.5 seconds. German cars tend to accelerate faster and have a slightly higher variation in acceleration times. Both types show that most cars fall between the 25th and 75th percentiles, but German cars generally perform better in terms of quicker acceleration.